pkg/segment/query/processor/searcher.go

```diff
@@ -171,6 +204,10 @@ func (s *Searcher) fetchRRCs() (*iqr.IQR, error) {
 	// we always take blocks from the front of the list.
 	s.remainingBlocksSorted = s.remainingBlocksSorted[len(nextBlocks):]
 
+	if len(s.remainingBlocksSorted) == 0 {
+		s.gotBlocks = false
+	}
```

Comment:
Why should we do this?

Reply:
I see. I think saying that in a comment would be good

Reply:
addressed in the latest commit.